pkg/workflow/workflowrun/pod_test.go

```diff
@@ -0,0 +1,458 @@
+									{
+										Name: "c1",
+									},
+								},
+							},
+						},
+					},
+				}, nil
+			case "unresolvable-argument":
+				return true, &v1alpha1.Stage{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: name,
+					},
+					Spec: v1alpha1.StageSpec{
+						Pod: &v1alpha1.PodWorkload{
+							Inputs: v1alpha1.Inputs{
+								Arguments: []v1alpha1.ArgumentValue{
+									{
+										Name: "undefined-arg",
+									},
+								},
+							},
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:  "c1",
+										Image: "{{ undefined-arg }}",
+									},
+								},
+							},
+						},
+					},
+				}, nil
+			case "stage1":
+				return true, &v1alpha1.Stage{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: name,
+					},
+					Spec: v1alpha1.StageSpec{
+						Pod: &v1alpha1.PodWorkload{
+							Inputs: v1alpha1.Inputs{
+								Arguments: []v1alpha1.ArgumentValue{
+									{
+										Name:  "image",
+										Value: "alpine:latest",
+									},
+									{
+										Name:  "dir",
+										Value: "/default",
+									},
+									{
+										Name:  "p1",
+										Value: "default1",
+									},
+								},
+								Resources: []v1alpha1.ResourceItem{
+									{
+										Name: "git",
+										Path: "/resource",
+									},
+								},
+							},
+							Outputs: v1alpha1.Outputs{
+								Artifacts: []v1alpha1.ArtifactItem{
+									{
+										Name: "art1",
+										Path: "/tmp/artifact.tar",
+									},
+								},
+							},
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:       "c1",
+										Image:      "{{ image }}",
+										WorkingDir: "{{ dir }}",
+									},
+								},
+							},
+						},
+					},
+				}, nil
+			case "stage2":
+				return true, &v1alpha1.Stage{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: name,
+					},
+					Spec: v1alpha1.StageSpec{
+						Pod: &v1alpha1.PodWorkload{
+							Inputs: v1alpha1.Inputs{
+								Resources: []v1alpha1.ResourceItem{
+									{
+										Name: "git-persistent",
+										Path: "/resource",
+									},
+								},
+								Artifacts: []v1alpha1.ArtifactItem{
+									{
+										Name: "art1",
+										Path: "/tmp/art1",
+									},
+								},
+							},
+							Outputs: v1alpha1.Outputs{
+								Resources: []v1alpha1.ResourceItem{
+									{
+										Name: "image",
+										Path: "/workspace/image.tar",
+									},
+								},
+							},
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:  "c1",
+										Image: "busybox:v1.0",
+									},
+								},
+							},
+						},
+					},
+				}, nil
+			default:
+				return true, nil, errors.NewNotFound(action.GetResource().GroupResource(), name)
+			}
+		}
+		return true, nil, nil
+	})
+	suite.client = client
+}
+
+func (suite *PodBuilderSuite) TestPrepare() {
+	builder := NewPodBuilder(suite.client, wf, wfr, "non-exist")
```

Comment:
Use following pattern to avoid duplicated code?

```suggestion
	stages := []string{"non-exist", "no-pod"}
	for _, stage := range stages {
		builder := NewPodBuilder(suite.client, wf, wfr, stage)
		assert.Error(suite.T(), builder.Prepare())
	}
```